libs/wingc/src/jsify.rs

```diff
@@ -655,107 +656,105 @@ impl<'a> JSifier<'a> {
 		}
 	}
 
-	pub fn jsify_struct_properties(
-		&self,
-		fields: &Vec<StructField>,
-		extends: &Vec<UserDefinedType>,
-		ctx: &mut JSifyContext,
-	) -> CodeMaker {
+	pub fn jsify_struct_env_properties(&self, env: &SymbolEnv, ctx: &mut JSifyContext) -> CodeMaker {
 		let mut code = CodeMaker::default();
-
-		// Any parents we need to get their properties
-		for e in extends {
-			code.line(format!(
-				"...require(\"{}\")().jsonSchema().properties,",
-				struct_filename(&e.root.name)
-			))
-		}
-
-		for field in fields {
+		for (field_name, (.., kind)) in env.symbol_map.iter() {
 			code.line(format!(
-				"{}: {{ {} }},",
-				field.name.name,
-				self.jsify_struct_field_to_json_schema_type(&field.member_type.kind, ctx)
+				"{}: {},",
+				field_name,
+				self.jsify_struct_schema_field(&kind.as_variable().unwrap().type_, ctx)
 			));
 		}
+		code
+	}
 
+	pub fn jsify_struct_schema_required_fields(&self, env: &SymbolEnv) -> CodeMaker {
```

Comment:
`self` doesn't need to be included as a parameter if it's not being used

Reply:
Oh my gewdness.. I thought it was required, but I guess thats just leftover Python-isms TIL

Reply:
Actually it seems like it does have to be included in oder to be callable from other methods within itself